src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,61 @@
+package duke.task;
+
+/**
+ * Represents a task.
+ * @author Tee Kok Siang
+ */
+public abstract class Task {
+    /** Description of the task */
+    protected String description;
+    /** Indicates if task is done */
+    protected boolean isDone;
+
+    /**
+     * Constructs a Task object.
+     *
+     * @param description Task description.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    private String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+    }
+
+    /**
+     * Marks task as done
+     */
+    public void markDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Returns formatted task information.
+     * It will be used to write into the file.
+     * @return Formatted task information.
+     */
+    public abstract String toFileString();
```

Comment:
I like this method's name - wish I had thought of this 😄

Should there should be a line space between the header and the parameters though? Can also consider this with the rest of the Javadocs in this class and its subclasses.

Reply:
Oh, yes. I missed that part of the comments section. Thanks for reminding me about it!